examples/advanced_operations/add_performance_max_campaign.py

```diff
@@ -120,6 +120,9 @@ def main(client, customer_id, audience_id):
         headline_asset_resource_names,
         description_asset_resource_names,
     )
+    asset_group_signal_operations = create_asset_group_signal_operations(
+        client, customer_id, audience_id
```

Comment:
Does this work if no audience_id is provided or will it throw an error?

Reply:
With my test it throws the following error when `audience_id` is `None`:
```
[2024-08-29 11:43:41 - WARNING] Request made: ClientCustomerId: REDACTED, Host: googleads.googleapis.com, Method: /google.ads.googleads.v17.services.GoogleAdsService/Mutate, RequestId: REDACTED, IsFault: True, FaultMessage: 'None' part of the resource name is invalid.
Request with ID "REDACTED" failed with status "NOT_FOUND" and includes the following errors:
Error with message "'None' part of the resource name is invalid.".
                On field: mutate_operations
                On field: asset_group_signal_operation
                On field: create
                On field: audience
                On field: audience
```

Reply:
Can you adjust the behavior so that you skip that part of the code if audience ID isn't provided, similar to how the example worked before?

Reply:
The `audience_id` check logic was moved to L650 below, within the function `create_asset_group_signal_operations`.

I think I misunderstood what you meant in the previous comment. The test was done by removing the `if audience_id:` check at L650 to observe what happens if `audience_id` is `None`. Running the complete updated example works without issues.